File: abc131-/abc133_c.py
l, r = map(int, input().split())
ans = 2019
cnt = min(l + 2019, r)
for i in range(l, cnt):
    for j in range(i + 1, cnt + 1):
        ans = min(ans, i*j % 2019)
print(ans)

File: abc131-/abc133_c.cpp
#include <bits/stdc++.h>
using namespace std;
using ll = long long;

#define repi(i, a, b) for (ll i = ll(a); i < ll(b); i++)

int main() {
    ll l, r;
    cin >> l >> r;

    ll ans = 2019, sup = min(r + 1, l + 2020);
    repi(i, l, sup) {
        repi(j, i + 1, sup) {
            ans = min(ans, i * j % 2019);
        }
    }
    cout << ans << endl;
    return 0;
}
